### PR TITLE
Activity List class incompatibility fix with return type

### LINKED ISSUE #423: [Bug]: Activity List class incompatibility with Core return type
### Expected behavior

image

### Actual behavior

image

### Steps to reproduce

On 11.x + CMF 4.x, there's a problem due the native return type

To solve it, probably just need class `CustomerManagementFrameworkBundle\Model\ActivityList\DefaultMariaDbActivityList\MySqlDao` to implement `DaoInterface`

### PR DESCRIPTION
**Pull request by aryaantony92**
Resolves #423

### CODE REVIEW

**Comment by aryaantony92** [comment]
Tests would be fixed by #432

File: src/Model/ActivityList/DefaultMariaDbActivityList/MySqlDao.php [hunk 1]
```diff
@@ -150,4 +152,13 @@ protected function addOrder(QueryBuilder $select)
...
         return $this;
     }
+
+    public function configure(): void
+    {
+    }
+
+    public function setModel(AbstractModel $model): static
+    {
+        return $this;
```
Comment (martineiber) [inline src/Model/ActivityList/DefaultMariaDbActivityList/MySqlDao.php:164]: When we use the setModel from DaoInterface, then we should to a right implementation and also set the model?

```suggestion
        $this->model = $model;
        
        return $this;
```

**Review by martineiber — CHANGES_REQUESTED** [review]
#432 is merged. 
Please rebase your branch to fix the failing tests.

File: src/Model/ActivityList/DefaultMariaDbActivityList/MySqlDao.php (hunk 1 above)
Comment (aryaantony92) [reply src/Model/ActivityList/DefaultMariaDbActivityList/MySqlDao.php:164]: Yes 👍

**Comment by martineiber** [comment]
@aryaantony92 
I was just thinking if we should add the interface to the MariaDbDao.php as well?
https://github.com/pimcore/customer-data-framework/blob/99da5e523e8435bed005c18304bc936c129c3de0/src/Model/ActivityList/DefaultMariaDbActivityList/MariaDbDao.php#L23

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -174,3 +174,13 @@ parameters:
 			message: "#^Call to an undefined method Pimcore\\\\Model\\\\DataObject\\\\Data\\\\ObjectMetadata\\:\\:getApplication_counter\\(\\)\\.$#"
 			count: 2
 			path: src/Targeting/ActionHandler/ApplyTargetGroupsFromSegments.php
+
+		-
+		    message: "#^Property CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\DefaultMariaDbActivityList\\\\MySqlDao::$model (CustomerManagementFrameworkBundle\Model\ActivityList\MySqlActivityList) does not accept Pimcore\\\\Model\\\\AbstractModel.$#"
```
Comment (martineiber) [inline phpstan-baseline.neon]: This should fix the stan error. 
```suggestion
		    message: "#^Property CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\DefaultMariaDbActivityList\\\\MySqlDao\\:\\:$model (CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\MySqlActivityList) does not accept Pimcore\\\\Model\\\\AbstractModel.$#"
```

**Review by martineiber — CHANGES_REQUESTED** [review]

File: phpstan-baseline.neon [hunk 3]
```diff
@@ -174,3 +174,13 @@ parameters:
 			message: "#^Call to an undefined method Pimcore\\\\Model\\\\DataObject\\\\Data\\\\ObjectMetadata\\:\\:getApplication_counter\\(\\)\\.$#"
 			count: 2
 			path: src/Targeting/ActionHandler/ApplyTargetGroupsFromSegments.php
+
+		-
+		    message: "#^Property CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\DefaultMariaDbActivityList\\\\MySqlDao::$model (CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\MySqlActivityList) does not accept Pimcore\\\\Model\\\\AbstractModel.$#"
```
Comment (martineiber) [inline phpstan-baseline.neon]:
```suggestion
		    message: "#^Property CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\DefaultMariaDbActivityList\\\\MySqlDao\\:\\:$model (CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\MySqlActivityList) does not accept Pimcore\\\\Model\\\\AbstractModel.$#"
```

File: phpstan-baseline.neon [hunk 4]
```diff
@@ -174,3 +174,13 @@ parameters:
...
+		    count: 1
+		    path: src/Model/ActivityList/DefaultMariaDbActivityList/MySqlDao.php
+
+		-
+		    message: "#^Property CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\DefaultMariaDbActivityList\\\\MariaDbDao::$model (CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\DefaultMariaDbActivityList) does not accept Pimcore\\\\Model\\\\AbstractModel.$#"
```
Comment (martineiber) [inline phpstan-baseline.neon]:
```suggestion
		    message: "#^Property CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\DefaultMariaDbActivityList\\\\MariaDbDao\\:\\:$model (CustomerManagementFrameworkBundle\\\\Model\\\\ActivityList\\\\DefaultMariaDbActivityList) does not accept Pimcore\\\\Model\\\\AbstractModel.$#"
```

**Review by martineiber — CHANGES_REQUESTED** [review]
let's try this change :D

**Comment by blankse** [comment]
You know you can create the baseline file via "-b" option?
https://phpstan.org/user-guide/command-line-usage#--generate-baseline%7C-b

**Comment by martineiber** [comment]
> You know you can create the baseline file via "-b" option? https://phpstan.org/user-guide/command-line-usage#--generate-baseline%7C-b

Didn't know that, thanks for the info :)

**Comment by martineiber** [comment]
Waiting for https://github.com/pimcore/customer-data-framework/pull/447

**Review by martineiber — APPROVED** [review]

**Pull request merged**